### ISSUE TITLE
How didSelect method can be implemented?

### ISSUE THREAD

**Issue by cresouls** [popei69/TemplateProject#4]
Hi, I am curious to know how UITableviewDelegate(didSelect) can be implemented in this pattern?

**Comment by ankitkushwah** [comment]
hii @cresouls  implement  UITableviewDelegate(didSelect) method in CurrencyDataSource class

**Comment by popei69** [comment]
@cresouls it really depends how you want to keep the separation of concern. 

For UI behaviour, I would personally implement `UITableViewDelegate` with the view, so into the `CurrencyViewController`.

For data logic, like add / edit / delete, maybe it makes more sense to leave it to the data layer, so into the `CurrencyDataSource`. 

If you feel the logic concerns the data as well as UI, maybe best it to leave into the viewModel layer, so the `CurrencyViewModel`.

In conclusion, there is not only _one_ solution, it's a matter to fit the best your purpose, keeping in mind a clean code easy to test following a clear separation of concern. Hope it helps!
